File: EncounterSimulator/Models/Encounter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EncounterSimulator.Models
{
    public class EncounterData
    {
        public int Id { get; set; } 

        public List<ActiveCharacter> Characters { get; set; }

        public TimeOfDay TimeOfEncounter { get; set; }

        public string Description { get; set; }

        public int TurnsElapsed { get; set; }

        public int RoundsElapsed { get; set; }

        public int SecondsElapsed { get; set; }

        public DateTime StartTime { get; set; }

        public DateTime EndTime { get; set; }

        public EncounterData() { }

        public EncounterData(int id, TimeOfDay timeOfDay, string description, int turnsElapsed, int roundsElapsed, int secondsElapsed, DateTime startTime, DateTime endTime)
        {
            Id = id;
            TimeOfEncounter = timeOfDay;
            Description = description;
            TurnsElapsed = turnsElapsed;
            RoundsElapsed = roundsElapsed;
            SecondsElapsed = secondsElapsed;
            StartTime = startTime;
            EndTime = endTime;
        }
    }

    public class Action
    {
        public int CharacterId { get; set; }

        public int TargetCharacterId { get; set; }

        public ActionType ActionType { get; set; }

        public double Value { get; set; }

        public List<Status> InflictedStatuses { get; set; }

        public string FlavorText { get; set; }

        public int EncounterId { get; set; }
    }

    public enum TimeOfDay
    {
        Unknown,
        Dawn,
        Morning,
        Afternoon,
        Evening,
        Dusk,
        Night
    }
}

File: EncounterSimulator/src/models/encounter.ts
﻿import { Status, ActionType } from "./game";
import { ActiveCharacter } from "./character";

export class EncounterData {
    id: number;

    characters: ActiveCharacter[];

    timeOfEncounter: TimeOfDay;

    description: string;

    roundsElapsed: number = 0;

    turnsElapsed: number = 0;

    secondsElapsed: number = 0;

    startTime: Date;

    endTime: Date;
}

// Describes one action that a character can take on their turn
export class Action {
    characterId: number;

    targetCharacterId: number;

    encounterId: number;

    actionType: ActionType;

    value: number;

    inflictedStatuses: Status[];

    flavorText: string;
}

export enum TimeOfDay {
    Dawn,
    Morning,
    Afternoon,
    Evening,
    Dusk,
    Night
}